Entities/Arm.cpp

```diff
@@ -552,24 +549,20 @@ void Arm::Update()
                 m_pHeldMO->SetHFlipped(m_HFlipped);
                 m_pHeldMO->SetRotAngle(m_Rotation.GetRadAngle());
                 m_pHeldMO->Update();
-                // If it blew up or whatever, releaes it from hand and put into scene so it'll be cleaned up properly
+                // If it blew up or whatever, release it from hand and put into scene so it'll be cleaned up properly
                 if (m_pHeldMO->IsSetToDelete())
                     g_MovableMan.AddItem(ReleaseHeldMO());
             }
         }
 
-// TODO: improve!")
         // Set correct frame for arm bend.
-        float halfMax = m_MaxLength / 2;
-        float balle = m_HandOffset.GetMagnitude() - halfMax;
-        float temp = (m_HandOffset.GetMagnitude() - halfMax) / halfMax;
-        temp *= m_FrameCount;
-        int newFrame = floorf(temp);
-        newFrame -= newFrame >= m_FrameCount ? 1 : 0;
-        m_Frame = newFrame;
-
-//        m_aSprite->SetAngle(m_Rotation);
-//        m_aSprite->SetScale(m_Scale);
+		float animProgress = (m_HandOffset.GetMagnitude() / m_MaxLength) * static_cast<float>(m_FrameCount);
```

Comment:
This seems a bit odd, why have animProgress at all if it's just used to set newFrame? Could maybe also get rid of the 2 casts that way too.